src/btree/bt_slvg.c

```diff
@@ -344,15 +347,21 @@ __wt_salvage(WT_SESSION_IMPL *session, const char *cfg[])
     /*
      * !!! (Don't format the comment.)
      * Step 7:
+     * Track the maximum write gen of the leaf pages and set that as the btree write gen.
      * Build an internal page that references all of the leaf pages, and write it, as well as any
      * merged pages, to the file.
      *
      * Count how many leaf pages we have (we could track this during the array shuffling/splitting,
      * but that's a lot harder).
      */
     for (leaf_cnt = i = 0; i < ss->pages_next; ++i)
-        if (ss->pages[i] != NULL)
+        if (ss->pages[i] != NULL) {
             ++leaf_cnt;
+            btree_write_gen = WT_MAX(btree_write_gen, ss->pages[i]->shared->gen);
+        }
+
+    btree->write_gen = btree_write_gen;
```

Comment:
Two questions:
1. does the local variable `btree_write_gen` add anything over just updating `btree->write_gen` directly in the loop?
2. in the case of metadata salvage, I don't think we need to set `conn->base_write_gen` explicitly here because `__wt_metadata_init_base_write_gen` will be called after this.  If that's correct, a comment here explaining that would help.

Reply:
I should have spelled it out more in q1 above: do we need to think about whether `btree->write_gen` could already be larger than the calculated `btree_write_gen`?

Reply:
Agreed. Removed the local variable and added a comment about connection base write gen.